File: common/src/main/java/scooterdelta/challenge/bot/common/domain/Weapon.java
package scooterdelta.challenge.bot.common.domain;

import scooterdelta.challenge.bot.common.lookup.WeaponType;

import java.util.Objects;

public class Weapon {

    private WeaponType weaponType;

    public WeaponType getWeaponType() {
        return weaponType;
    }

    public void setWeaponType(final WeaponType weaponType) {
        this.weaponType = weaponType;
    }

    @Override
    public String toString() {
        return "Weapon{" +
                "weaponType=" + weaponType +
                '}';
    }

    @Override
    public boolean equals(final Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        final Weapon weapon = (Weapon) o;
        return weaponType == weapon.weaponType;
    }

    @Override
    public int hashCode() {
        return Objects.hash(weaponType);
    }
}

File: common/src/main/kotlin/scooterdelta/challenge/bot/common/state/remote/domain/Weapon.kt
package scooterdelta.challenge.bot.common.state.remote.domain

import com.fasterxml.jackson.annotation.JsonProperty
import scooterdelta.challenge.bot.common.lookup.WeaponType

data class Weapon(

        @JsonProperty("WeaponType")
        val weaponType: WeaponType,

        @JsonProperty("EnergyRequired")
        val energyRequired: Int
)